docs/DeveloperGuide.adoc

```diff
@@ -592,31 +592,80 @@ https://docs.oracle.com/javafx/2/api/javafx/scene/image/Image.html
 ** Pros: Less memory consumption.
 ** Cons: Original Image may be lost.
 
-=== Emailing a contact
+
+diagrams::ModelComponentClassDiagram.pptx[width="650"]
+
+// tag::email[]
+
+=== Emailing a contact using Google API
 
 ==== Current implementation
-The `EmailCommand` class will allow user to open an email interface to send email to the chosen contact. `EmailCommand` extends the `Command` class. It takes in an index through the `EmailCommandParser`,
-similar to the implementation of `DeleteCommand` and `DeleteCommandParser`.
+The sending of email will be implemented using the Google API.
+
+Firstly, to access the Google API, the user must login to Google using the `googlelogin` command.
+The `GoogleLoginCommand` class will open the Google authentication webpage in the built-in web browser for user to login.
+
+The `GoogleAuthentication` class will handle the authentication process of Google login.
+After authentication is successful, the user will be redirected to the Google homepage,
+where the `Token` is stored in the URL.
+
+Below is the sequence diagram of the `googlelogin` command:
+
+.Sequence Diagram of the GoogleLoginCommand
+image::GoogleLoginSequence.png[width="600", align="center"]
+
+Next, a `GmailClient` is created using the `GoogleAuthentication` class and `Token`.
+The `Token` is obtained from the redirected URL, meaning that the user must not navigate to other URL
+after they have logged in.
 
-The sending of email will be implemented using the Gmail API.
+Below is the class diagram of the `GmailClient`:
 
-Firstly, `GoogleAuthorizationCodeFlow` is used to build and request for authorization from user.
-Then it saves the user authorization into `Credential`, which is saved into a local file so that they do not need to repeat the authorization step.
-This authorization is handled in the `GmailAuthentication` class.
+.Class Diagram of GmailClient and GoogleAuthentication
+image::GmailClassDiagram.png[width="600", align="center"]
 
-Next, a `GmailClient` is created using the `Credential` that was created by `GmailAuthentication`.
-The `GmailClient` class provides all the services that uses the Gmail API.
```

Comment:
use

Reply:
@trafalgarandre which one you referring to